### ISSUE TITLE
Use wayland and sway

### ISSUE THREAD

**Issue by luisdavim** [regolith-linux/regolith-desktop#27]
Sway is a drop in replacement for i3 on wayland how hard would it be to migrate?

**Comment by kgilmer** [comment]
That's a great question!  I do not know much about Wayland, and would expect there to be some deep changes in the gnome integration. If that even makes sense in Wayland I can't say.  I opted to target X11 for application support and general familiarity.  I would guess that the UI themeing stuff would be pretty straight forward, but there may be dragons lurking in lower integration levels.

**Comment by muniter** [comment]
I agree with @kgilmer, sway is a fantastic project with super potential, I'm an eager user of i3 and have tried it, but I myself think It would be good to wait for a little more Wayland maturity before attempting to set up something like regolith on it.

**Comment by kgilmer** [comment]
I'll add the feedback wanted tag and maybe someone with a deeper understanding of Wayland could determine the level of effort.

**Comment by kgilmer** [comment]
#61 needs to be completed before work on this can begin.

**Comment by shadowrylander** [comment]
From Hacker News [`Regolith Linux – A desktop with less visual bullshit`](https://news.ycombinator.com/item?id=19702266), is what user [`ac130kz`](https://news.ycombinator.com/user?id=ac130kz) said technically true? 
> Install Ubuntu minimal, build sway, install the packages they are using for system configuration and basically you're done

**Comment by kgilmer** [comment]
Hi @shadowrylander , I cannot say as I do not have any experience with sway.  There are some parts of Regolith that are compatibly with Wayland/Sway such as Rofi, but other programs may not be, or may require different configuration.  One issue is that color and font data is moving to Xresources (in Regolith) and I'm told that Wayland does not have a similar facility.

In any case, I don't think you'd be able to just install packages and have it work.  You would need to extract the files and modify them as needed.  If you'd like to try applying Regolith theme stuff to Wayland, I'll try and help as best I can.

**Comment by shadowrylander** [comment]
Thanks @kgilmer! That makes sense. At the moment, I need a bit more experience with modifying `i3` themes in general, but if I try, I'll be sure to ask!

**Comment by kgilmer** [comment]
Regolith relies on Xresources for configuration, which is not available in wayland.  Based on this I do not see how Regolith could be ported easily.  A better option would be to create something new that is Wayland-first in terms of configuration and principles.

**Comment by appelgriebsch** [comment]
Hey, I did some test-runs with sway on ManjaroARM. It turns out in sway you can configure variables directly in the sway configuration files. It also allows you to separate sections of the file and use an include direction to put the different parts together...

As a reference here you can see how you can configure different styles / colors via variables: https://github.com/rkubosz/base16-sway

Here you can see how you can split the file into different sections merging them together via includes: https://github.com/robertjk/dotfiles/blob/253b86442dae4d07d872e8b963fa33b5f8819594/.config/sway/config

**Comment by Adam-Antios** [comment]
Hello! Is there any update on this? Also I would like to ask, when you say that a better option would be the creation of something new, does that mean a completely new project with different name and goals or the re-write of Regolith from the ground up? It may be a minor issue at this point but Regolith as a brand and as a project has gained attention, without mentioning how amazing the name and logo are!

**Comment by kgilmer** [comment]
Thanks for the kind words @Adam-Antios.  The Regolith "experience" and branding may be easily integrated into Wayland/Sway, I'm not sure.  I guess I was just playing it safe with my earlier message.  Given that the non-modularity of the i3 config seems to be a substantial issue for users during upgrades, it may make sense to do some more investigation into Sway in the near tearm.

**Comment by kgilmer** [comment]
FYI, work in this area will be documented here: https://github.com/orgs/regolith-linux/projects/8

**Comment by Adam-Antios** [comment]
Hey! What you say is completely understandable. From what I see most teams haven't really put a lot of work migrating to Wayland yet. I just wanted to see what the state of the project is. Glad that you are monitoring the situation!

**Comment by hypergig** [comment]
....take it from me, sway is not ready for prime time _just_ yet. Especially if you are on a newer hidpi display. Or if you are a corporate user and need zoom to work. Or if you are a big jetbrains user.

I switched from sway to regolith a while back and keep track of the issues here https://github.com/hypergig/workstation/pull/2

**Comment by ploum** [comment]
Ubuntu 21.04 will default on Wayland.

**Comment by ploum** [comment]
This post could be of interest to @kgilmer : someone on Sway created a rofi-clone by "simply" opening a terminal without border with fzf launched. Very smart.

gemini://ew.srht.site/en/2022/20220112-debian-on-hifive-unmatched-2.gmi
